docs: improve guides (#3066)
webpack/webpack.js.org: src/content/guides/production.md

  - mehrdaad
  - SevenOutman
  - AnayaDesign
  - wizardofhogwarts
---

In this guide, we'll dive into some of the best practices and utilities for building a production site or application.

T> This walkthrough stems from [Tree Shaking](/guides/tree-shaking) and [Development](/guides/development). Please ensure you are familiar with the concepts/setup introduced in those guides before continuing on.

## Setup

> The goals of _development_ and _production_ build differ greatly. In _development_, we want strong source mapping and a localhost server with live reloading or hot module replacement. In _production_, our goals shift to a focus on minified bundles, lighter weight source maps, and optimized assets to improve load time. With this logical separation at hand, we typically recommend writing __separate webpack configurations__ for each environment.

Reviewer: one and two (builds), why this change?

Author: According to subject verb agreement, `build` will fit into this context.

Author: @EugeneHlushko @montogeek please check it again

  - mehrdaad
  - SevenOutman
  - AnayaDesign
  - wizardofhogwarts
  - aholzner
---

In this guide, we'll dive into some of the best practices and utilities for building a production site or application.

T> This walkthrough stems from [Tree Shaking](/guides/tree-shaking) and [Development](/guides/development). Please ensure you are familiar with the concepts/setup introduced in those guides before continuing on.

## Setup

> The goals of _development_ and _production_ build differ greatly. In _development_, we want strong source mapping and a localhost server with live reloading or hot module replacement. In _production_, our goals shift to a focus on minified bundles, lighter weight source maps, and optimized assets to improve load time. With this logical separation at hand, we typically recommend writing __separate webpack configurations__ for each environment.

Reviewer: I still think that `development build and production build` becomes `builds`.
If it was saying: `the difference between those builds` you wouldn't change it to `build` right?

I did take a look at this: https://www.grammarbook.com/grammar/subjectVerbAgree.asp
But still cannot see why you want to change that 😅

Author: I was talking in context of goals. But no worries if you cross checked it. I may revert it if u say.

Reviewer: or easier apply suggest 
```suggestion
The goals of _development_ and _production_ builds differ greatly. In _development_, we want strong source mapping and a localhost server with live reloading or hot module replacement. In _production_, our goals shift to a focus on minified bundles, lighter weight source maps, and optimized assets to improve load time. With this logical separation at hand, we typically recommend writing __separate webpack configurations__ for each environment.
```

## Source Mapping

> We encourage you to have source maps enabled in production, as they are useful for debugging as well as running benchmark tests. That said, you should choose one with a fairly quick build speed that's recommended for production use (see [`devtool`](/configuration/devtool)). For this guide, we'll use the `source-map` option in _production_ as opposed to the `inline-source-map` we used in the _development_:

Reviewer: so preceding `in production` looks good to you, but in development should receive a `the` ? I don't get it

Author: @EugeneHlushko I must have missed it while reading. Adding in next commit.

## CLI Alternatives

> Some of what has been described above are also achievable via the command line. For example, the `--optimize-minimize` flag will include the `TerserPlugin` behind the scenes. The `--define process.env.NODE_ENV="'production'"` will do the same for the `DefinePlugin` instance described above. And, `webpack -p` will automatically invoke both those flags and thus the plugins to be included.

Reviewer: ```suggestion
Some of what has been described above can also be achieved by using the command line. For example, the `--optimize-minimize` flag will include the `TerserPlugin` behind the scenes. The `--define process.env.NODE_ENV="'production'"` will do the same for the `DefinePlugin` instance described above. And, `webpack -p` will automatically invoke both those flags and thus the plugins to be included.
```
